uint8 code
uint8 direction
uint8 magnitude
uint8 turn
uint8 accel